corehq/apps/reports_core/static/reports_core/js/maps.js

```diff
@@ -56,14 +56,16 @@ hqDefine('reports_core/js/maps', function () {
     var initPopupTemplate = function (config) {
         if (!privates.template || !_.isEqual(config.columns, privates.columns)) {
             privates.columns = config.columns;
-            var rows = _.map(privates.columns, function (col) {
-                var tr = _.template("<tr><td><%- label %></td>")(col);
-                tr += "<td><%- " + col.column_id + "%></td></tr>";
-                return tr;
-            });
-            var table = '<table class="table table-bordered"><%- rows %></table>';
-            var template = _.template(table)({rows: rows.join('')});
-            privates.template = _.template(template);
+            privates.template = _.template([
+                "<table class='table table-bordered'>",
+                "   <% _.map(columns, function (col) { %>",
+                "     <tr>",
+                "       <td><%- col.label %></td>",
+                "       <td><%- row[col.column_id] %></td>",
```

Comment:
Was this an intentional change from the previous code? Previously I think it was `<%- " + col.column_id + "%>`

Reply:
Yeah it's intentional. Check out my now-outdated comment above, but basically the whole `"<%- " + col.column_id + "%>"` approach is pretty unsafe—you're injecting user input into template code. (Imagine `col.column_id` is `"alert('hello')"`—whatever JS expression you put there will just get run.) So I tried to reverse engineer the intent of that and it's basically, let's say `col.column_id` is `"date_of_birth"`, then the template it would create is `<%- date_of_birth %>`. What's the template context that has a "date_of_birth" field? I looked below in another function where it gets rendered as "privates.template(row)". Then I changed that to the more explicit `privates.template({row: row})`, so that instead of `<%- date_of_birth %>` I can use `<%- row['date_of_birth'] %>`. And then finally instead of the string literal 'date_of_birth' appearing in the template at all, we replace it an outer forloop within the template that pulls from `columns` which gets passed in to the template context, which is updated to `privates.template({row: row, columns: columns})`. So in short, it looks different because the approach being used before was convoluted, but it does the same thing.